### DISCUSSION TITLE
Stoping a manual Topology scan

### DISCUSSION THREAD

**Discussion by pipiche38** [zigpy/zigpy#1377] [Q&A] [answered]
we are starting a manual Topology scan by ` await self.topology.scan()`

however if we want to stop it  `self.topology._scan_task.cancel()`seems to have no effect

**Comment by puddly** [comment]
What version of Python are you using?

**Reply by pipiche38** [reply, marked as answer]
Python3.10.11

But I guess that is on my side.

**Reply by pipiche38** [reply]
My fault. @puddly one question or comment. When zigpy start, then if configured (which is default) the TOP_SCAN will launch the periodic tasks, but first it will start to wait for the period.

This mean that after a start, the topology will be updated only 4 hours after the start. Wouldn't it make sense to do the first much sooner, and then wait for the period ?

**Reply by dmulcahey** [reply]
Not all devices are immediately addressable at startup. The network needs time to stabilize, for routes to report, etc. And we already do a lot of stuff at startup. Adding an immediate topo scan isn’t worth it IMO.

**Comment by MattWestb** [comment]
Most Zigbee 3 SEDs is doing checkens every 55 minute so they is coming online in one hour if the system / coordinator was down for more then one hour. If like have the topography faster its no use then the routers have not seen there children and its not having all in its neighbor table.
Going to pre Zigbee and not certificated device you never knowing then there is talking with its parents so not knowing then they is being online or not and with time frame they is using for pulling its parent.
